rotkehlchen/exchanges/ftx.py

```diff
@@ -0,0 +1,514 @@
+import hmac
+from http import HTTPStatus
+import logging
+import time
+from collections import defaultdict
+from json.decoder import JSONDecodeError
+from typing import TYPE_CHECKING, Any, DefaultDict, Dict, List, Optional, Tuple, Union, overload
+from urllib.parse import urlencode
+
+import gevent
+import requests
+from typing_extensions import Literal
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.assets.converters import asset_from_ftx
+from rotkehlchen.assets.utils import symbol_to_asset_or_token
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors import DeserializationError, RemoteError, UnknownAsset, UnsupportedAsset
+from rotkehlchen.exchanges.data_structures import AssetMovement, MarginPosition, Trade
+from rotkehlchen.exchanges.exchange import ExchangeInterface, ExchangeQueryBalances
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_asset_amount_force_positive,
+    deserialize_fee,
+    deserialize_price,
+    deserialize_timestamp_from_date,
+    deserialize_trade_type,
+)
+from rotkehlchen.typing import (
+    ApiKey,
+    ApiSecret,
+    AssetMovementCategory,
+    Fee,
+    Location,
+    Timestamp,
+)
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.interfaces import cache_response_timewise, protect_with_lock
+from rotkehlchen.utils.serialization import jsonloads_dict
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+INITIAL_BACKOFF_TIME = 2
+BACKOFF_LIMIT = 60
+PAGINATION_LIMIT = 100
+
+
+def trade_from_ftx(raw_trade: Dict[str, Any]) -> Optional[Trade]:
+    """Turns an FTX transaction into a rotki Trade.
+
+    Mary raise:
+        - UnknownAsset due to Asset instantiation
+        - DeserializationError due to unexpected format of dict entries
+        - KeyError due to dict entires missing an expected entry
+    """
+    # In the case of perpetuals and futures this fields can be None
+    if raw_trade.get('baseCurrency') is None:
+        return None
+    if raw_trade.get('quoteCurrency') is None:
+        return None
```

Comment:
```suggestion
    if raw_trade.get('baseCurrency', None) is None:
        return None
    if raw_trade.get('quoteCurrency', None) is None:
        return None
```

A `get()` without a default argument is the same as `raw_trade['baseCurrency']`.

Reply:
Maybe I'm missing something here but

```
>>> a = {}
>>> a.get('a') is None
True
```

Reply:
Hmm you are correct. I wonder why I was under the impression it would still raise a KeyError. Ignore my comment then! Perhaps my tendency to have everything explicit. But in this case it's not needed indeed.